src/h5web/visualizations/heatmap/DomainSlider.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import ReactSlider from 'react-slider';
+import shallow from 'zustand/shallow';
+import { format } from 'd3-format';
+import styles from './DomainSlider.module.css';
+import { useHeatmapStore, Domain } from './store';
+
+const EXTEND_PERCENTAGE = 0.2;
+const MIN_DECIMALS = 1;
+
+function round(value: number, decimals: number): number {
+  return Number(`${Math.round(Number(`${value}e${decimals}`))}e-${decimals}`);
+}
+
+function DomainSlider(): JSX.Element {
+  const [customDomain, dataDomain, setCustomDomain] = useHeatmapStore(state => [
+    state.customDomain,
+    state.dataDomain,
+    state.setCustomDomain,
+    shallow,
+  ]);
+
+  if (dataDomain === undefined) {
+    return <></>;
+  }
+
+  const domainExtent = dataDomain[1] - dataDomain[0];
+  const domainExtension = EXTEND_PERCENTAGE * domainExtent;
```

Comment:
An extension to the "true" domain so that slider values can go above/below the true domain.

Reply:
Interesting. Do you think it's a feature people might want?

Reply:
If so, perhaps it would be simpler to store this extended domain in the store directly.

Reply:
Yes it is helpful sometimes to alleviate some saturation effects. :+1: for the extended domain.